R/expect_null_linter.R

```diff
@@ -0,0 +1,48 @@
+#' expect_null Linter
+#'
+#' Require usage of expect_null(x) over expect_equal(x, NULL) and similar
+#'   [testthat::expect_null()] exists specifically for testing for `NULL` objects.
+#'   [testthat::expect_equal()], [testthat::expect_identical()], and
+#'   [testthat::expect_true()] can also be used for such tests,
+#'   but it is better to use the tailored function instead.
+#'
+#' @evalRd rd_tags("expect_null_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+expect_null_linter <- function() {
```

Comment:
as parameters here we could potentially toggle which of the cases apply:

(1) `expect_equal(x, NULL)` ➡️  `expect_null(x)`
(2) `expect_identical(x, NULL)` ➡️  `expect_null(x)`
(3) `expect_true(is.null(x))` ➡️  `expect_null(x)`

Is it worth doing so?

Reply:
I think it's fine, or are there legitimate use-cases / differences in behavior for the other forms?

Reply:
not really. just trying to imagine a user wanting finer control. I guess we can wait for an FR.

Reply:
Agreed.